src/main/java/ru/yandex/qatools/ashot/comparison/PointsMarkupPolicy.java

```diff
@@ -88,21 +86,28 @@ protected void markDiffPoints(BufferedImage image) {
 
     private Set<Point> getDeposedPoints() {
         if (deposedPoints.isEmpty()) {
-            deposedPoints = deposeReference(this);
+            deposedPoints = deposeReference();
         }
         return deposedPoints;
     }
 
-    private Point getReferenceCorner(PointsMarkupPolicy diff) {
-        double x = min(diff.diffPoints, on(Point.class).getX());
-        double y = min(diff.diffPoints, on(Point.class).getY());
+    private Point getReferenceCorner() {
+        Iterator<Point> iterator = diffPoints.iterator();
+        Point diffPoint = iterator.next();
```

Comment:
Previous code assumed that `diffPoints` is always non-empty because empty collection led to exception at unboxing since lambdaj method returned `null`